src/components/shared/Activity/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { ReactNode } from "react";
+import { Member } from "../../../store/reducers/members";
+import { CurrencyValue } from "../Currency";
+import { RouteName } from "../Navigation";
+
+/**
+ * Represents the direction of the relationship between actors in an activity.
+ * Currently only supports onward or bidirectional, since that makes the flow
+ * of the application linear and therefore more comprehensible.
+ */
+export enum ActivityDirection {
+  Forward = "Forward",
+  Bidirectional = "Bidirectional"
+}
+
+/**
+ * The content of an Activity. Missing some metadata that makes a complete,
+ * renderable Activity.
+ */
+export interface ActivityContent {
+  actor: Member;
+  description: (string | CurrencyValue)[];
+  body: ReactNode;
```

Comment:
Ah perhaps this ought to be agnostic of React, like everything else is. Basically the content body types are icon, set of videos/images, or post text based on my mockups.

Reply:
done